distributed/tests/test_priorities.py

```diff
@@ -51,48 +57,40 @@ async def block_worker(
     """
     if pause:
         w.status = Status.paused
-        while s.workers[w.address].status != Status.paused:
-            await asyncio.sleep(0.01)
+        await async_wait_for(
+            lambda: s.workers[w.address].status == Status.paused, timeout=5
+        )
     else:
         ev = Event()
-        clog = c.submit(lambda ev: ev.wait(), ev, key="block_worker")
-        while "block_worker" not in w.state.tasks:
-            await asyncio.sleep(0.01)
+        clog = c.submit(ev.wait, key="block_worker")
+        await wait_for_state(clog.key, "executing", w)
 
     yield
 
     if ntasks_on_scheduler is None:
         ntasks_on_scheduler = len(c.futures)
     if ntasks_on_worker is None:
         ntasks_on_worker = len(c.futures)
-    while len(s.tasks) < ntasks_on_scheduler:
-        await asyncio.sleep(0.01)
+    await async_wait_for(lambda: len(s.tasks) >= ntasks_on_scheduler, timeout=5)
 
     if pause:
-        assert len(s.unrunnable) == ntasks_on_worker
+        assert (
+            len(s.unrunnable) == ntasks_on_worker or len(s.queued) == ntasks_on_worker
+        )
         assert not w.state.tasks
         w.status = Status.running
     else:
-        while len(w.state.tasks) < ntasks_on_worker:
-            await asyncio.sleep(0.01)
         await ev.set()
         await clog
         del clog
-        while "block_worker" in s.tasks:
-            await asyncio.sleep(0.01)
+        await async_wait_for(lambda: "block_worker" not in s.tasks, timeout=5)
 
 
 def gen_blockable_cluster(test_func):
     """Generate a cluster with 1 worker and disabled memory monitor,
     to be used together with ``async with block_worker(...):``.
     """
-    return pytest.mark.parametrize(
-        "pause",
-        [
-            pytest.param(False, id="queue on worker"),
-            pytest.param(True, id="queue on scheduler"),
-        ],
```

Comment:
Why remove the ids?

Reply:
They weren't accurate anymore. I could call then `pause` and `clog` or something.